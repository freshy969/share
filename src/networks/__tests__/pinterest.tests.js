import { when } from 'mobx';
import Pinterest from '../pinterest';

describe('Share > Pinterest', () => {
  test('url', () => {
    const pinterest = Pinterest.create({});
    pinterest.entityLink = jest.fn();
    pinterest.entityLink.mockReturnValueOnce('https://demo.frontity.com/the-beauties-of-gullfoss');

    expect(
      pinterest.url({
        type: 'post',
        id: 60,
        media: 'https://demo.frontity.com/wp-content/uploads/2016/11/Iceland-test-1200x726.jpg',
        description:
          'Gullfoss is a waterfall located in the canyon of the Hvítá river in southwest Iceland.',
      }),
    ).toMatchSnapshot();
  });

  test('requestCount success', done => {
    const request = { get: jest.fn() };
    const pinterest = Pinterest.create({}, { request });
    pinterest.entityLink = jest.fn();
    pinterest.entityLink.mockReturnValueOnce('https://demo.frontity.com/the-beauties-of-gullfoss');

    request.get.mockResolvedValueOnce({
      text: 'cb({"url":"https://demo.frontity.com/the-beauties-of-gullfoss","count":123})',
    });

    when(
      () => pinterest.count({ type: 'post', id: 60 }),
      () => {
        expect(pinterest.count({ type: 'post', id: 60 })).toBe(123);
        expect(pinterest.entityLink.mock.calls).toMatchSnapshot();
        expect(pinterest.entityLink.mock.results).toMatchSnapshot();
        expect(request.get.mock.calls).toMatchSnapshot();
        expect(request.get.mock.results).toMatchSnapshot();
        done();
      },
    );

    pinterest.requestCount({ type: 'post', id: 60 });
  });

  test('requestCount fails', async () => {
    const request = { get: jest.fn() };
    const pinterest = Pinterest.create({}, { request });
    pinterest.entityLink = jest.fn();
    pinterest.entityLink.mockReturnValue('https://demo.frontity.com/the-beauties-of-gullfoss');

    // First request (request fails)
    request.get.mockRejectedValueOnce('fail');
    await pinterest.requestCount({ type: 'post', id: 60 });
    expect(pinterest.count({ type: 'post', id: 60 })).toBe(null);

    // Second request (bad response)
    request.get.mockResolvedValueOnce({ bad: 'response' });
    await pinterest.requestCount({ type: 'post', id: 60 });
    expect(pinterest.count({ type: 'post', id: 60 })).toBe(null);

    expect(request.get.mock.calls).toMatchSnapshot();
  });
});
