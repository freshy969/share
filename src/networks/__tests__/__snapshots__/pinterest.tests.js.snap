// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Share > Pinterest requestCount fails 1`] = `
Array [
  Array [
    "https://api.pinterest.com/v1/urls/count.json?cb=cb&url=https%3A%2F%2Fdemo.frontity.com%2Fthe-beauties-of-gullfoss",
  ],
  Array [
    "https://api.pinterest.com/v1/urls/count.json?cb=cb&url=https%3A%2F%2Fdemo.frontity.com%2Fthe-beauties-of-gullfoss",
  ],
]
`;

exports[`Share > Pinterest requestCount success 1`] = `
Array [
  Array [
    Object {
      "id": 60,
      "type": "post",
    },
  ],
]
`;

exports[`Share > Pinterest requestCount success 2`] = `
Array [
  Object {
    "isThrow": false,
    "value": "https://demo.frontity.com/the-beauties-of-gullfoss",
  },
]
`;

exports[`Share > Pinterest requestCount success 3`] = `
Array [
  Array [
    "https://api.pinterest.com/v1/urls/count.json?cb=cb&url=https%3A%2F%2Fdemo.frontity.com%2Fthe-beauties-of-gullfoss",
  ],
]
`;

exports[`Share > Pinterest requestCount success 4`] = `
Array [
  Object {
    "isThrow": false,
    "value": Promise {},
  },
]
`;

exports[`Share > Pinterest url 1`] = `"https://pinterest.com/pin/create/button?url=https%3A%2F%2Fdemo.frontity.com%2Fthe-beauties-of-gullfoss&media=https%3A%2F%2Fdemo.frontity.com%2Fwp-content%2Fuploads%2F2016%2F11%2FIceland-test-1200x726.jpg&description=Gullfoss%20is%20a%20waterfall%20located%20in%20the%20canyon%20of%20the%20Hv%C3%ADt%C3%A1%20river%20in%20southwest%20Iceland."`;
